whenchanged/whenchanged.py

```diff
@@ -135,49 +137,62 @@ def run(self):
 def print_usage(prog):
     print(__doc__ % {'prog': prog}, end='')
 
+def parse_args(argv):
+    """
+    Parse the various args and returns a dictionnary.
 
-def main():
-    args = sys.argv
-    prog = os.path.basename(args.pop(0))
-
-    if '-h' in args or '--help' in args:
-        print_usage(prog)
-        exit(0)
+    The function returns None if the arguments were asking for the help.
 
+    The function raises ValueError in case of bad arguments.
+    """
     files = []
     command = []
     recursive = False
     verbose = False
     run_once = False
     run_at_start = False
 
-    while args and args[0][0] == '-':
-        flag = args.pop(0)
-        if flag == '-v':
+    prog = os.path.basename(argv.pop(0))
+
+    # pre-parsing
+    # -c is a very special case: all following arguments are part of a command and should not be parsed
+    argv_until_c = argv
+    for i, a in enumerate(argv):
+        m = re.match(r'\-[^\-c]*c(.?)', a)
```

Comment:
I still don't like this way of handling `-c`. This use case is really common though, just look at `xargs`, `gdb --args`, `find -exec`, etc. So I'm a bit surprised that getopt doesn't support it. What it does seem to support, though, is the special `--` option which stops parsing:

```
In [23]: gnu_getopt("-rs -- cmd -s".split(), 'hvr1s')
Out[23]: ([('-r', ''), ('-s', '')], ['cmd', '-s'])
```

So what about replacing `-c` with `--` before parsing?